transport/spi/src/main/kotlin/EndpointComponent.kt

```diff
@@ -80,6 +82,14 @@ abstract class EndpointComponent<T : Any>(
                 delay(CHECK_INTERVAL_SECONDS * 1000)
             }
         }
+
+        suspend fun generateKeepAliveFile() =
+            withContext(Dispatchers.IO) {
+                val file = File(System.getProperty("user.home"), KEEP_ALIVE_FILE_NAME)
```

Comment:
Suggestion: This code to locate the lock file is also used when checking for its presence. Maybe it can be extracted to a helper function.